tui-markdown/src/options.rs

```diff
@@ -0,0 +1,40 @@
+//! Rendering configuration for tui-markdown.
+//!
+//! For now the only knob is the theme [`StyleSheet`].  The struct purposefully
+//! remains `#[non_exhaustive]` *in spirit* (not in syntax to avoid MSRV
+//! bumps): we recommend construction through the builder pattern or the field
+//! shorthand with `..Default::default()` so that adding new options becomes a
+//! non-breaking change.
+
+use crate::style_sheet::{BuiltinStyleSheet, StyleSheet};
+
+/// Collection of optional parameters that influence markdown rendering.
+///
+/// The generic `S` allows users to carry their own theme type without forcing
+/// dynamic dispatch.  Libraries that do not want the extra generic in their
+/// call-chain can always box it (`Box<dyn StyleSheet>`).
+#[derive(Clone)]
+pub struct Options<S: StyleSheet = BuiltinStyleSheet> {
+    /// The [`StyleSheet`] implementation that will be consulted every time the
+    /// renderer needs a colour choice.
```

Comment:
nit: Here and elsewhere, use the US spelling of color. I'm Australian (currently in the US), but I generally use the US spelling regardless on programming terms like this because it's easier to be consistent across docs and code.

```suggestion
    /// renderer needs a color choice.
```